moztelemetry/dataset.py

```diff
@@ -156,10 +156,8 @@ def summaries(self, limit=None):
 
         with futures.ProcessPoolExecutor(MAX_CONCURRENCY) as executor:
             scanned = self._scan(schema, [self.prefix], clauses, executor)
```

Comment:
I don't remember the numbers, but in most of the cases the long pole is listing the keys. It really depends on the filter though, the extreme cases (i.e. either no filters or only one filter on buildid) behave very differently.